core/trino-parser/src/main/java/io/trino/sql/ReservedIdentifiers.java

```diff
@@ -44,6 +44,27 @@
 
     private static final SqlParser PARSER = new SqlParser();
 
+    private static final Set<String> SQL_KEYWORDS;
+
+    private static final Set<String> RESERVED_IDENTIFIERS;
+
+    static {
+        ImmutableSet.Builder<String> names = ImmutableSet.builder();
+        Vocabulary vocabulary = SqlBaseLexer.VOCABULARY;
+        for (int i = 0; i <= vocabulary.getMaxTokenType(); i++) {
+            String name = nullToEmpty(vocabulary.getLiteralName(i));
+            Matcher matcher = IDENTIFIER.matcher(name);
+            if (matcher.matches()) {
+                names.add(matcher.group(1));
+            }
+        }
+        SQL_KEYWORDS = names.build();
+        RESERVED_IDENTIFIERS = SQL_KEYWORDS.stream()
+                .filter(ReservedIdentifiers::reserved)
+                .sorted()
+                .collect(toImmutableSet());
+    }
```

Comment:
This change should go in a separate commit. It's tangentially related to the main fix in this PR (needed for performance), but it's not an essential part of the change.

Reply:
Moved this to #13739

Reply:
@nevillelyh let's have this as separate commit, but please keep this within this PR.

Reply:
@findepi done

Reply:
Can you please also change the order of commits? `Make reservedIdentifiers & sqlKeywords static` should be first.
Also, add some explanation there, like "This makes reservedIdentifiers() and sqlKeywords() constant time operations."

Reply:
Fixed!

Reply:
thanks